tutorials/plot_dics.py

```diff
@@ -0,0 +1,272 @@
+# -*- coding: utf-8 -*-
+"""
+DICS for power mapping
+======================
+
+In this tutorial, we're going to simulate two signals originating from two
+locations on the cortex. These signals will be sine waves, so we'll be looking
+at oscillatory activity (as opposed to evoked activity).
+
+We'll be using dynamic imaging of coherent sources (DICS) [1]_ to map out
+spectral power along the cortex. Let's see if we can find our two simulated
+sources.
+"""
+# Author: Marijn van Vliet <w.m.vanvliet@gmail.com>
+#
+# License: BSD (3-clause)
+
+###############################################################################
+# Setup
+# -----
+# We first import the required packages to run this tutorial and define a list
+# of filenames for various things we'll be using.
+import os.path as op
+import numpy as np
+from scipy.signal import welch, coherence
+from mayavi import mlab
+from matplotlib import pyplot as plt
+
+import mne
+from mne.datasets import sample
+from mne.minimum_norm import make_inverse_operator, apply_inverse
+from mne.time_frequency import csd_epochs
+from mne.beamformer import make_dics, apply_dics_csd
+
+# Suppress irrelevant output
+mne.set_log_level('ERROR')
+
+# We use the MEG and MRI setup from the MNE-sample dataset
+data_path = sample.data_path(download=False)
+subjects_dir = op.join(data_path, 'subjects')
+mri_path = op.join(subjects_dir, 'sample')
+
+# Filenames for various files we'll be using
+meg_path = op.join(data_path, 'MEG', 'sample')
+trans_fname = op.join(meg_path, 'sample_audvis_raw-trans.fif')
+fwd_fname = op.join(meg_path, 'sample_audvis-meg-eeg-oct-6-fwd.fif')
+
+# Seed for the random number generator
+np.random.seed(42)
```

Comment:
isn't this evil? I suspect the following parts of the code don't get affected but I would still try not to advertise it

Reply:
Isn't this standard practice for examples so that the results reproduce exactly every time? Of course when doing real analysis, you would never do this.

Reply:
I usually advocate using `rng = np.random.RandomState(42)` instead to avoid global states altogether, maybe that is what @jasmainak is referring to

Reply:
Ah, of course! That makes sense.